core/server/services/mega/post-email-serializer.js

```diff
@@ -4,21 +4,65 @@ const settingsCache = require('../../services/settings/cache');
 const urlUtils = require('../../lib/url-utils');
 const moment = require('moment');
 const cheerio = require('cheerio');
+const api = require('../../api');
 
 const getSite = () => {
     return Object.assign({}, settingsCache.getPublic(), {
         url: urlUtils.urlFor('home', true)
     });
 };
 
-const serialize = (post) => {
+/**
+ * createUnsubscribeUrl
+ *
+ * Takes a member and returns the url that should be used to unsubscribe
+ * In case of no member, generates the preview unsubscribe url - `?preview=1`
+ *
+ * @param {object} member
+ * @param {string} member.uuid
+ */
+const createUnsubscribeUrl = (member) => {
```

Comment:
None of the other options except `uuid` are used in this method. Just passing it directly would do I think ;) Also would avoid having to pass empty object when you want a previewURL. The check would just become `if (uuid) {set uuid in the url} else {preview parameter}

Reply:
Yeah good point, didn't zoom out enough to clean this properly, will apply the changes 😄